test/openssl/test_pkey_ec.rb

```diff
@@ -241,7 +241,23 @@ def test_ECPrivateKey_encrypted
     0/dGSU5SzFG+iT9iFXCwCvv+bxyegkBOyALFje1NAsM=
     -----END EC PRIVATE KEY-----
     EOF
-    key = OpenSSL::PKey::EC.new(pem, "abcdef")
+    begin
+      key = OpenSSL::PKey::EC.new(pem, "abcdef")
+    rescue OpenSSL::PKey::ECError => e
+      raise e unless OpenSSL.fips_mode
+
+      # The password based encryption used in the PEM format uses MD5 for
+      # deriving the encryption key from the password, and MD5 is not
+      # FIPS-approved.
+      #
+      # See https://github.com/openssl/openssl/discussions/21830#discussioncomment-6865636
+      # for details.
+      #
+      # The error message is not the best. But it is expected.
+      assert_equal "OpenSSL::PKey::ECError", e.class.name
+      assert_equal "invalid curve name", e.message
+      return
+    end
```

Comment:
Can we just do `return if OpenSSL.fips_mode` at the beginning of the method?

Testing the error message doesn't make much sense. I might even say the error message "invalid curve name" is a borderline bug.

[It doesn't seem worth fixing, though, considering that `OpenSSL::PKey::EC.new(curve_name_string)` is almost completely useless on OpenSSL >= 3.0 due to pkeys having become immutable.]

Reply:
I think the same change as this will be needed for other pkey test files, as I understand we aim to make the same `rake test` run successfully.

Reply:
> Can we just do `return if OpenSSL.fips_mode` at the beginning of the method?

Yes, we can. I will fix it.

> Testing the error message doesn't make much sense. I might even say the error message "invalid curve name" is a borderline bug.
>
> [It doesn't seem worth fixing, though, considering that `OpenSSL::PKey::EC.new(curve_name_string)` is almost completely useless on OpenSSL >= 3.0 due to pkeys having become immutable.]

OK. I can understand it.

Reply:
> I think the same change as this will be needed for other pkey test files, as I understand we aim to make the same `rake test` run successfully.

Do you think we can include the fixes for the other pkey test files in this PR?

I was thinking to fix the `test_pkey_ec.rb` first, then other pkey test files `test_Pkey.*.rb` next in another PR(s). But I will take a look at the other pkey test files if I can fix it in this PR.

Reply:
> I want to fix the other `test_pkey*.rb` files in another PR.

Sounds good!